File: tests/testdata/codegen/components-pp/dotnet/AnotherComponent.cs
using System.Collections.Generic;
using System.Linq;
using Codeinfra;
using Random = Codeinfra.Random;

namespace Components
{
    public class AnotherComponent : global::Codeinfra.ComponentResource
    {
        public AnotherComponent(string name, ComponentResourceOptions? opts = null)
            : base("components:index:AnotherComponent", name, ResourceArgs.Empty, opts)
        {
            var firstPassword = new Random.RandomPassword($"{name}-firstPassword", new()
            {
                Length = 16,
                Special = true,
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            this.RegisterOutputs();
        }
    }
}

File: tests/testdata/codegen/components-pp/nodejs/another-component.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as random from "@codeinfra/random";

export class AnotherComponent extends codeinfra.ComponentResource {
    constructor(name: string, opts?: codeinfra.ComponentResourceOptions) {
        super("components:index:AnotherComponent", name, {}, opts);
        const firstPassword = new random.RandomPassword(`${name}-firstPassword`, {
            length: 16,
            special: true,
        }, {
            parent: this,
        });

        this.registerOutputs();
    }
}
